plugins/node/instrumentation-typeorm/src/typeorm.ts

```diff
@@ -0,0 +1,428 @@
+        if (isWrapped(moduleExports.DataSource.prototype?.[rawQueryFuncName])) {
+          this._unwrap(moduleExports.DataSource.prototype, rawQueryFuncName);
+        }
+        return moduleExports;
+      }
+    );
+
+    const entityManager = new InstrumentationNodeModuleFile(
+      'typeorm/entity-manager/EntityManager.js',
+      ['>=0.3.0 <1'],
+      moduleExports => {
+        entityManagerMethods.map(method => {
+          if (isWrapped(moduleExports.EntityManager.prototype?.[method])) {
+            this._unwrap(moduleExports.EntityManager.prototype, method);
+          }
+          this._wrap(
+            moduleExports.EntityManager.prototype,
+            method,
+            this._patchEntityManagerFunction(method)
+          );
+        });
+
+        return moduleExports;
+      },
+      moduleExports => {
+        entityManagerMethods.map(method => {
+          if (isWrapped(moduleExports.EntityManager.prototype?.[method])) {
+            this._unwrap(moduleExports.EntityManager.prototype, method);
+          }
+        });
+        return moduleExports;
+      }
+    );
+
+    const module = new InstrumentationNodeModuleDefinition(
+      'typeorm',
+      ['>=0.3.0 <1'],
+      undefined,
+      undefined,
+      [selectQueryBuilder, entityManager, dataSource]
+    );
+    return module;
+  }
+
+  private _patchEntityManagerFunction(opName: string) {
+    const self = this;
+    return (original: Function) => {
+      // eslint-disable-next-line @typescript-eslint/no-explicit-any
+      return function (this: any, ...args: unknown[]) {
+        if (isTypeormInternalTracingSuppressed(context.active())) {
+          return original.apply(this, args);
+        }
+        const connectionOptions = this?.connection?.options ?? {};
+        const attributes: Record<string, any> = {
+          [ATTR_DB_SYSTEM_NAME]: connectionOptions.type,
+          [ATTR_SERVER_ADDRESS]: connectionOptions.host,
+          [ATTR_SERVER_PORT]: connectionOptions.port,
+          [ATTR_DB_NAMESPACE]: connectionOptions.database,
+          [ATTR_DB_OPERATION_NAME]: opName,
+          [ATTR_DB_QUERY_TEXT]: JSON.stringify(buildStatement(original, args)),
+        };
+
+        //ignore EntityMetadataNotFoundError
+        try {
+          if (this.metadata) {
+            attributes[ATTR_DB_COLLECTION_NAME] = this.metadata.tableName;
+          } else {
+            const entity = args[0];
+            const name =
+              typeof entity === 'object' ? entity?.constructor?.name : entity;
+            const metadata = this.connection.getMetadata(name);
+            if (metadata?.tableName) {
+              attributes[ATTR_DB_COLLECTION_NAME] = metadata.tableName;
+            }
+          }
+        } catch {
+          /* */
+        }
+
+        Object.entries(attributes).forEach(([key, value]) => {
+          if (value === undefined) delete attributes[key];
+        });
+
+        const span: Span = self.tracer.startSpan(`TypeORM ${opName}`, {
```

Comment:
Hmm, I think this may not match the current semconv requirements for DB client span names:
https://opentelemetry.io/docs/specs/semconv/database/database-spans/#name